### PR TITLE
support manual tracer and custom reporter

### LINKED ISSUE #81: Providing manual report
It provides manual reporting of data to Zipkin, for example, what does a timer need to report data?

### PR DESCRIPTION
**Pull request by glmapper**
### Result:

Fixes  #81 #249

### CODE REVIEW

File: sofa-tracer-plugins/sofa-tracer-flexible-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <dependencies>
+
+        <dependency>
+            <groupId>com.alipay.sofa</groupId>
+            <artifactId>tracer-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.25</version>
```
Comment (ujjboy) [inline sofa-tracer-plugins/sofa-tracer-flexible-plugin/pom.xml]: remove version

File: sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/FlexibleStatJsonReporter.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ *
+ * @author: guolei.sgl (guolei.sgl@antfin.com) 2019/8/2 11:50 AM
+ * @since:
+ **/
+public class FlexibleStatJsonReporter extends AbstractSofaTracerStatisticReporter {
+
+    /***
+     * print builder
+     */
+    private static JsonStringBuilder statJson = new JsonStringBuilder();
```
Comment (ujjboy) [inline sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/FlexibleStatJsonReporter.java]: is JsonStringBuilder thread-safe?

File: sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/annotations/Tracer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * @author: guolei.sgl (guolei.sgl@antfin.com) 2019/8/2 11:39 AM
+ * @since:
+ **/
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD })
+public @interface Tracer {
+    /**
+     * 操作名
```
Comment (ujjboy) [inline sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/annotations/Tracer.java]: change to English comment.

**Review by ujjboy — CHANGES_REQUESTED** [review]

File: sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/FlexibleStatJsonReporter.java (hunk 2 above)
Comment (glmapper) [reply sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/FlexibleStatJsonReporter.java]: > is JsonStringBuilder thread-safe?
According to the component segregation

Comment (ujjboy) [reply sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/FlexibleStatJsonReporter.java]: so it's not thread-safe.  maybe not `static`

**Review by ujjboy — APPROVED** [review]
LGTM

**Pull request merged**